servlet.py

```diff
@@ -561,6 +565,7 @@ def get(self):
                         self.send_error(400, explanation='Invalid file type %s' % mtype)
         else:
             self.send_error(400, explanation='That pair is not installed')
+            yield doc_pipe_sem.release()
```

Comment:
NameError: name 'max_doc_pipes' is not defined

Please test your code before you ask for reviews.